invenio/modules/documents/utils.py

```diff
@@ -88,12 +88,23 @@ def _get_legacy_bibdoc(recid, filename=None):
     """
     from invenio.ext.login import current_user
     from invenio.legacy.bibdocfile.api import BibRecDocs
-    paths = [
+    return [
         (bibdoc.fullpath, bibdoc.is_restricted(current_user))
         for bibdoc in BibRecDocs(recid).list_latest_files(list_hidden=False)
         if not bibdoc.subformat and not filename or
         bibdoc.name + bibdoc.superformat == filename
```

Comment:
> How can one retrieve a file having a subformat?

I don't know.

> Can this filter be changed e.g. to:
> `if not filename or bibdoc.get_fullname() == fullname`?

Probably it can. I took a work from @drjova